sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -522,7 +522,8 @@ trait ArraySortLike extends ExpectsInputTypes {
         } else if (o2 == null) {
           nullOrder
         } else {
-          -ordering.compare(o1, o2)
+          // avoid false positive of RV_NEGATING_RESULT_OF_COMPARETO by SpotBugs
+          -1 * ordering.compare(o1, o2)
```

Comment:
Are we sure this Comparator never returns Int.MinValue? If it can, then this is still a legitimate possible bug.

Reply:
Good catch. We cannot ensure it. `ordering.compare(o2, o1)` looks better.